src/Subscriber/PaymentSubscriber.php

```diff
@@ -380,58 +401,49 @@ private function trans(string $snippet, array $parameters = []): string
 
-        if ($stateData) {
-            //Convert the state data into an array
-            $stateDataArray = json_decode($stateData, true);
+        //Convert the state data into an array
+        $stateDataArray = json_decode($stateData, true);
 
-            if (json_last_error() !== JSON_ERROR_NONE) {
-                $this->logger->error('Payment state data is an invalid JSON: ' . json_last_error_msg());
-                $this->session->getFlashBag()
-                    ->add('danger', $this->trans('adyen.paymentMethodSelectionError'));
-                return;
-            }
-
-            //Payment method selected from the Shopware methods form
-            $selectedPaymentMethod = $this->paymentMethodRepository->search(
-                (new Criteria())
-                    ->addFilter(new EqualsFilter('id', $event->getRequestDataBag()->get('paymentMethodId'))),
-                Context::createDefaultContext()
-            )->first();
+        if (json_last_error() !== JSON_ERROR_NONE) {
+            $this->logger->error('Payment state data is an invalid JSON: ' . json_last_error_msg());
+            $this->session->getFlashBag()
+                ->add('danger', $this->trans('adyen.paymentMethodSelectionError'));
+            return;
+        }
 
-            $selectedPaymentMethodIsStoredPM =
-                $selectedPaymentMethod->getFormattedHandlerIdentifier() == 'handler_adyen_oneclickpaymentmethodhandler';
+        $selectedPaymentMethodIsStoredPM =
+            $selectedPaymentMethod->getFormattedHandlerIdentifier() == 'handler_adyen_oneclickpaymentmethodhandler';
 
-            $stateDataIsStoredPM = !empty($stateDataArray["paymentMethod"]["storedPaymentMethodId"]);
+        $stateDataIsStoredPM = !empty($stateDataArray["paymentMethod"]["storedPaymentMethodId"]);
 
-            //Only store the state data if it matches the selected PM
-            if ($stateDataIsStoredPM == $selectedPaymentMethodIsStoredPM) {
-                try {
-                    $this->paymentStateDataService->insertPaymentStateData(
-                        $event->getSalesChannelContext()->getToken(),
-                        $event->getRequestDataBag()->get('adyenStateData'),
-                        $event->getRequestDataBag()->get('adyenOrigin')
-                    );
-                } catch (AdyenException $exception) {
-                    $this->session->getFlashBag()
-                        ->add('danger', $this->trans('adyen.paymentMethodSelectionError'));
-                    return;
-                }
-            } else {
-                //PM selected and state.data don't match, clear previous state.data
-                $this->paymentStateDataService->deletePaymentStateDataFromContextToken(
-                    $event->getSalesChannelContext()->getToken()
+        //Only store the state data if it matches the selected PM
+        if ($stateDataIsStoredPM == $selectedPaymentMethodIsStoredPM) {
```

Comment:
not really relevant here but just wondering any reason why you only use `==` and not `===`? the second one ensures the type matches too. I am mentioning this because e.g. `'1' == 1` results in `true` but `'1' === 1` results in `false`. The `===` is also a bit faster as it does not need to convert types.